src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -46,17 +49,20 @@ internal abstract class BrokerClient : IBrokerClient {
 
         public string Name { get; }
         public Uri Uri { get; }
+        public string RCommandLineArguments { get; }
         public bool IsRemote => !Uri.IsFile;
         public bool IsVerified { get; protected set; }
 
-        protected BrokerClient(string name, Uri brokerUri, string interpreterId, ICredentialsDecorator credentials, IActionLog log, IConsole console) {
+        protected BrokerClient(string name, Uri brokerUri, string rCommandLineArguments, string interpreterId, ICredentialsDecorator credentials, IActionLog log, IConsole console) {
             Name = name;
             Uri = brokerUri;
             Log = log;
 
+            RCommandLineArguments = rCommandLineArguments?.Trim() ?? string.Empty;
             _interpreterId = interpreterId;
             _credentials = credentials;
             _console = console;
+            _parametersHash = CalculateAdditionalParametersHash();
```

Comment:
It is already a combination of session id and `_parametersHash`. As you mentioned, `_parametersHash` is basically a connection id.

The problem with guid as a connection id is that it is random. To find matching connection, you have to either fetch data about all existing connections and find the right one, or ask broker to do it by providing all the parameters. In both cases, it requires changes in broker code and may have potential versioning issues (if we add or remove parameter). 
With hash instead of guid, it is guaranteed that for the same parameters you'll get the same id, and broker doesn't have to know anything about those parameters if they are VS or RHost specific.